src/components/ModalWithForm.js

```diff
@@ -21,6 +26,14 @@ export default class ModalWithForm extends Modal {
     super.close();
   }
 
+  renderLoading(isLoading) {
+    if (isLoading) {
+      this._saveButton.textContent = "Saving...";
+    } else {
+      this._saveButton.textContent = "Save";
+    }
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`

Reply:
The set up of the setInputValues function you've suggested looks very similar to my "_getInputValues" function and I'm not quite sure where in index.js to utilize it.
<img width="434" alt="Screen Shot 2024-08-10 at 1 24 54 AM" src="https://github.com/user-attachments/assets/0e28cb08-c9ff-4506-b683-7ae58cb56676">
Here? 
<img width="474" alt="Screen Shot 2024-08-10 at 1 26 05 AM" src="https://github.com/user-attachments/assets/279a96fb-0f89-4174-9052-9e1c358b2f4c">

Reply:
> Here?

Yes, on the profile form opening